### PR TITLE
#2089 - use snapcraft-preload to rewrite the /dev/shm path dynamically

### PR DESCRIPTION
**Pull request by c-kr**

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -35,8 +36,19 @@ parts:
...
       chmod +x $SNAPCRAFT_PART_INSTALL/bin/sabnzbd-wrapper
+      ln -sf ../usr/lib/libsnapcraft-preload.so $SNAPCRAFT_PART_INSTALL/lib/libsnapcraft-preload.so
     prime: [-usr/bin/7z,-usr/bin/7za,-usr/bin/7zr]
     organize:
       usr/bin/unrar-nonfree: usr/bin/unrar
+  snapcraft-preload:
+    source: https://github.com/sergiusens/snapcraft-preload.git
+    plugin: cmake
+    cmake-parameters:
+      - -DCMAKE_INSTALL_PREFIX=/usr -DLIBPATH=/lib
```
Comment (Safihre) [inline snap/snapcraft.yaml:49]: I'm a bit of a Linux/Cmake novice, but can't we change this parameter so the `ln -sf` a few lines up isn't needed anymore?

Comment (c-kr) [reply snap/snapcraft.yaml:49]: I tried this a few times and spend a few hours, as others did: https://forum.snapcraft.io/t/snapcraft-preload-and-dev-shm-access/28200/ but wasnt able to find a solution (tried DCMAKE_INSTALL_PREFIX=/ but didnt work). Mabye someone else has a better way to handle this?

**Comment by Safihre** [comment]
Ok, I'll just merge and see what happens!

**Comment by Safihre** [comment]
@c-kr Got build errors on armhf:
```
Failed to fetch stage packages: Error downloading packages for part 'snapcraft-preload': The package 'lib32stdc++6' was not found..
```
and arm64:
```
Could not find a required package in 'build-packages': g++-multilib
```

**Comment by Safihre** [comment]
https://github.com/sergiusens/snapcraft-preload/issues/13
Seems to have a solution.

**Pull request merged**